### PR TITLE
Report Operational Outcomes in Responses

### PR DESCRIPTION
**Pull request by jeremytwfortune**
## Description of Changes

It was previously very difficult, particularly in Python, to determine why an operation that returned a 400 status rejected the payload. The detailed reasons are often reported in OperationOutcomes in the response body, but these were being swallowed during exception handling.

This adds an exception layer to the SDK, exposing those classes, and also provides better exposure for those outcomes.

## Security

**REMINDER: All file contents are public.**

- [x] I have ensured no secure credentials or sensitive information remain in code, metadata, comments, etc. Of particular note: No temporary testing changes committed such as API base URLs, access tokens, print/log statements, etc.
- [x] My changes do not introduce any security risks, or any such risks have been properly mitigated.

This changes some output from the SDK, but these outcomes were previously already available to the client: this just wraps them in clear exceptions that can be handled and inspected.

There are new JSON and XML parsers to handle this, but all of the inspection is performed on responses from the server.

## Reviewers

- [x] I have assigned the appropriate reviewer(s).

### CODE REVIEW

File: python/orchestrate/_internal/http_handler.py [hunk 1]
```diff
@@ -1,8 +1,15 @@
-import os
 import json
+import os
+from dataclasses import dataclass
 from typing import Any, Mapping, Optional
+from xml.etree import ElementTree
```
Comment (jwyderko) [inline python/orchestrate/_internal/http_handler.py:5]: I think dependabot takes a dim view of xml.etree and maybe the defusedxml library works just as well here?

File: python/orchestrate/_internal/http_handler.py [hunk 2]
```diff
@@ -25,6 +32,90 @@ def _get_additional_headers() -> Mapping[str, str]:
...
+    outcomes = _read_json_outcomes(response)
+    if outcomes:
+        return [str(outcome) for outcome in outcomes]
+
+    return []
+
+
+def _exception_from_response(response: requests.Response) -> OrchestrateHttpError:
+    operational_outcomes = _read_operational_outcomes(response)
+    if str(response.status_code).startswith("4"):
```
Comment (jwyderko) [inline python/orchestrate/_internal/http_handler.py:114]: I believe Orchestrate may return some 50x responses with useful operational outcomes to consider for return to the client.

File: python/orchestrate/_internal/http_handler.py (hunk 1 above)
Comment (jeremytwfortune) [reply python/orchestrate/_internal/http_handler.py:5]: Hm. I'm not sure what you mean here. Is there a security advisory out on the stdlib?

File: python/orchestrate/_internal/http_handler.py (hunk 2 above)
Comment (jeremytwfortune) [reply python/orchestrate/_internal/http_handler.py:114]: Oh, that's good to know. Do you have any examples of this you can DM me?

File: python/orchestrate/_internal/http_handler.py (hunk 1 above)
Comment (jeremytwfortune) [reply python/orchestrate/_internal/http_handler.py:5]: Well, that's quite a conundrum. Introduce a third party dependency to keep dependabot at bay or keep getting annoyed at dependabot telling us that the XML data we're parsing is trusted?

Maybe what I should do instead here is just to return the full response text, unparsed?

**Review by jwyderko — APPROVED** [review]

**Pull request merged**